homeassistant/components/sensor/vera.py

```diff
@@ -185,3 +203,99 @@ def update(self):
+            'state_variable',
+            self.select_state_variable())
+        self._unit_of_measurement = self._config.get(
+            'unit_of_measurement',
+            self.select_units())
+        self._vera_device_id = self._config.get(
+            'vera_id',
+            self._device_data.get('id'))
+        self._parent_entity_id = discovery_info.get('parent_entity_id', None)
+
+    def select_state_variable(self):
+        """ Selects the variable to be used for the state of the parent device.
+        A child sensor that monitors this variable will not be created. """
+        # If no state variable is configured choose one.
+        # Put the variables in order of preference.
+        watchable_variables = [
+            'temperature',
+            'light',
+            'tripped',
+            'batterylevel',
+            'light',
+            'armed',
+            'humidity',
+            'armedtripped']
+
+        for val in watchable_variables:
+            if val in self._device_data.keys():
+                # See if this field is excluded for the device
+                if not self._config.get(val, {}).get('exclude', False):
+                    return val
+
+        return 'id'
+
+    def select_units(self):
+        """ Derive the unit of measurement for this sensor based on the
+        variable it uses for state. """
+        if self._state_variable == 'temperature':
+            # pylint: disable=unused-variable
+            temp, units = self.get_temperature()
+            return units
+        elif self._state_variable == 'light':
+            return 'lux'
+        elif self._state_variable == 'batterylevel':
+            return '%'
+        elif self._state_variable == 'humidity':
+            return '%'
+
+        return None
+
+    @property
+    def unit_of_measurement(self):
+        """ Unit of measurement of this entity, if any. """
+        return self._unit_of_measurement
+
+    @property
+    # pylint: disable=too-many-return-statements
+    def state(self):
+        """ Return the state value for this device """
+        if self._state_variable == 'temperature':
+            # pylint: disable=unused-variable
+            temp, units = self.get_temperature()
+            return temp
+        elif self._state_variable == 'armed':
+            if str(self._device_data.get('armed')) == '1':
+                return STATE_ARMED
+            else:
+                return STATE_NOT_ARMED
+        elif self._state_variable == 'tripped':
+            if str(self._device_data.get('tripped')) == '1':
+                return STATE_TRIPPED
+            else:
+                return STATE_NOT_TRIPPED
+        elif self._state_variable == 'armed_tripped':
+            if str(self._device_data.get('armedtripped')) == '1':
+                return STATE_TRIPPED
+            else:
+                return STATE_NOT_TRIPPED
+
+        return self._device_data.get(self._state_variable, None)
```

Comment:
Instead of returning `None` if the variable is not there, `STATE_UNKNOWN` from `homeassistant.const` might be more appropriate.